internal/request/request.go

```diff
@@ -0,0 +1,200 @@
+package request
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/catalogtask"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/common"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/jsonwriter"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/logger"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/tarwriter"
+	"github.com/RedHatInsights/catalog_mqtt_client/internal/towerapiworker"
+	mqtt "github.com/eclipse/paho.mqtt.golang"
+	log "github.com/sirupsen/logrus"
+)
+
+type PageWriter interface {
+	Write(name string, b []byte) error
+	Flush() error
+	FlushErrors(msg []string) error
```

Comment:
@bzwei Do you want to add Cleanup() here

Reply:
we can look at it in a future PR